### ISSUE TITLE
Cant access preferences

### ISSUE THREAD

**Issue by trwww** [RealRaven2000/FiltaQuilla#37]
Access to FiltaQuilla preferences have seemed to have disappeared.

The add-on auto-updated, and I'm on the latest ThunderBird, and I can't find preferences any more.

<img width="784" alt="Screen Shot 2019-12-17 at 1 19 27 PM" src="https://user-images.githubusercontent.com/216991/71023385-a0209e00-20d0-11ea-9f7f-bd34f1ab7d22.png">

**Comment by RealRaven2000** [comment]
All preferences of add-ons are in the menu Tools - Add-on Options. ... since Thunderbird 60. What version did you update from? The options button in the Add-on Manager is long gone!

**Comment by trwww** [comment]
Yes thats where I looked for it at first and originally set up the options. But that submenu is empty now.

<img width="773" alt="Screen Shot 2019-12-17 at 2 49 05 PM" src="https://user-images.githubusercontent.com/216991/71029021-7d948200-20dc-11ea-868a-36200a62fa51.png">

The reason I mentioned the options button instead of the menu is because thats where this says to set the options:

http://www.mesquilla.net/mesquillacom/extensions/filtaquilla/index.html

**Comment by RealRaven2000** [comment]
> 
> <img alt="Screen Shot 2019-12-17 at 2 49 05 PM" width="773" src="https://user-images.githubusercontent.com/216991/71029021-7d948200-20dc-11ea-868a-36200a62fa51.png">

Looks like all your Add-ons were broken or disabled by the Thunderbird update. Can you update them:
![image](https://user-images.githubusercontent.com/6314674/71030677-d3335380-2109-11ea-8d33-0ac942456782.png)
...and then disable one by one? It's most likely a broken Add-on pretending to be compatible, that's causing this.

> The reason I mentioned the options button instead of the menu is because thats where this says to set the options:
> 
> http://www.mesquilla.net/mesquillacom/extensions/filtaquilla/index.html

this link (and site) is outdated, where did you find that?

The official support site is here: [quickfilters.mozdev.org/filtaquilla.html](http://quickfilters.mozdev.org/filtaquilla.html)

**Comment by trwww** [comment]
>  It's most likely a broken Add-on pretending to be compatible, that's causing this.

Unfortunately the only two add-ons I have installed are this one and the lightning/calendar.

> this link (and site) is outdated, where did you find that?

By typing in 'FiltaQuilla configure options' in to google. Its the first result.

Also, not sure what you mean by "Looks like all your Add-ons were broken or disabled by the Thunderbird update." The screenshot I provided shows them enabled. But I did 'Check for Updates' like you suggested. Everything is up to date, like I mentioned when I opened the ticket.

<img width="947" alt="Screen Shot 2019-12-17 at 3 23 16 PM" src="https://user-images.githubusercontent.com/216991/71031643-c7339b80-20e1-11ea-8f68-f445686712bb.png">

Is there a command I can run in the console that will open the preferences? I tried to google to find something, but I can't find it.

The extension is definitely loaded and running - I see the extension provided options in the 'Perform these actions' dropdown in the filters UI.

Thanks for the awesome extension!

**Comment by RealRaven2000** [comment]
> Also, not sure what you mean by "Looks like all your Add-ons were broken or disabled by the Thunderbird update." The screenshot I provided shows them enabled. 

hmm, really makes me wonder what broke...

> But I did 'Check for Updates' like you suggested. Everything is up to date, like I mentioned when I opened the ticket.

thunderbird 68 had a bug not updating them automatically.

> Is there a command I can run in the console that will open the preferences? I tried to google to find something, but I can't find it.

It's in thunderbird options, advanced, general tab, config editor button:
![image](https://user-images.githubusercontent.com/6314674/71032221-ce23d380-210c-11ea-9f60-3ff77da00078.png)
 
> The extension is definitely loaded and running - I see the extension provided options in the 'Perform these actions' dropdown in the filters UI.
> 

then you need to check the error log (Tools / deveoper tools / error console) for errors on startup or when you open the Add-ons submenu (not sure when Tb populates it) for some hints. And you can also check what happens when you open the filter editor.

**Comment by trwww** [comment]
> then you need to check the error log (Tools / deveoper tools / error console) for errors on startup

Ah, yeah theres something there:

ReferenceError: window is not defined | filtaquilla-util.js:642:11

> And you can also check what happens when you open the filter editor.

This seems to work fine:

FiltaQuilla 15:45:59.639  [224260 ms] fq_FilterEditor.js - start...
FiltaQuilla 15:45:59.639  [0 ms] fq_FilterEditor.js - Finished.

<img width="669" alt="Screen Shot 2019-12-17 at 3 46 25 PM" src="https://user-images.githubusercontent.com/216991/71033067-a882d400-20e4-11ea-9dd0-43c353e6791d.png">

Thanks again!

**Comment by RealRaven2000** [comment]
Ok. Can you unpack and install the following version please:

[FiltaQuilla-2.1pre7.zip](https://github.com/RealRaven2000/FiltaQuilla/files/3975635/FiltaQuilla-2.1pre7.zip)

**Comment by trwww** [comment]
I used 'Install Add-on from file' and TB said it wasn't compatible:

<img width="445" alt="Screen Shot 2019-12-17 at 5 18 47 PM" src="https://user-images.githubusercontent.com/216991/71039121-6613c400-20f1-11ea-85d4-734ac4ad3299.png">

I first uninstalled the old one and restarted a couple times to make sure it was completely uninstalled.

**Comment by trwww** [comment]
The ReferenceError exception is gone, but I still can't access preferences through 'Tools > Add-on preferences'.

There is a new error, but I don't see FiltaQuilla in that call stack.

<img width="1354" alt="Screen Shot 2019-12-17 at 5 27 07 PM" src="https://user-images.githubusercontent.com/216991/71039688-be979100-20f2-11ea-8118-2fa85959e497.png">

**Comment by trwww** [comment]
To add, I created a blank Thunderbird profile and added the .pre7.xpi to it, and get the same result - no Add-on preferences, with similar output in the debugging console. I'd dig in to the code but I'm lost :(

<img width="1275" alt="Screen Shot 2019-12-17 at 5 41 28 PM" src="https://user-images.githubusercontent.com/216991/71040520-bb050980-20f4-11ea-80f1-197003525cfb.png">

**Comment by RealRaven2000** [comment]
hmm, I have no idea what is happening here either. none of the errors seem related. ALso I do not know what Thunderbird does internally when it adds the options linkl I do think it would scan manifest.json for the "options" section. I don't think any of the Add-on Javascript code is executed at this stage; but Syntax errors or internal errors / exceptions in Thunderbird might lead to the code never being executed. Every window has its own global variables and menus, so you might want to try opening a folder in a new window and see if any other error messages come up.

**Comment by trwww** [comment]
Sure. Thanks for taking a look at it. I installed a couple arbitrary extensions in the Trash profile and everything works as expected. I'll back up my main profile that I have filters in and poke stuff hard, but don't have much time for it at the moment. It was all working fine before the TB 68 release. Thanks again!

**Comment by RealRaven2000** [comment]
Is your Thunderbird version an official one from thunderbird.net? I assume you are using a Mac (personally I don't have any apple devices) but I know there are some problem with packaged (internal) extensions in Linux / Debian packages.

**Comment by trwww** [comment]
Yes, its its the "official" thunderbird app

**Comment by RealRaven2000** [comment]
I am currently out of office (for the last 8 days) expecting to be back in action after the Christmas holidays, if you find any new logs or want to ping me as a reminder, just leave another comment below. thanks very much!

**Comment by jason-hub** [comment]
I have the same problem!  Since updating to TB 68.3.1, the preferences menu is completely blank.  I disabled FiltaQuilla, and the problem persists.

**Comment by Dimathman** [comment]
Have just updated to TB 68.3.1. And have updated add-ons, including QuickFilters Pro.
I also have this problem where the preferences for FiltaQuilla have gone missing, exactly as per all the dialogue and responses in the dialogue above - as if it were me contributing rather than trwww and jason-hub. FiltaQuilla is installed and working with the default options avaiable, but it is now impossible to change anything.

**Comment by RealRaven2000** [comment]
> 
> 
> Have just updated to TB 68.3.1. And have updated add-ons, including QuickFilters Pro.

Make sure you really have FiltaQuilla 2.0 and QF 4.17.5 installed. There is a bug in Thunderbird that prevents proper Add-on update after upgrading to Tb68. Often it's best to manually "install over".

The installed version is shown in the top right:

![image](https://user-images.githubusercontent.com/6314674/72094317-1601d900-330e-11ea-909c-a5dc68d41845.png)

If you're still missing items in the tools > Addons submenus, check what other Add-ons are installed, some of them are capable of breaking it.

**Comment by Dimathman** [comment]
I completely removed and reinstalled both FiltaQuilla and QuickFilters Pro, in that order. FiltaQuilla still showed no preferences on install; that is, before reinstalling QuickFilters. Subsequently, QuickFilters Pro reinstalled OK, back to normal, but still no preferences for FiltaQuilla.

Puzzling though, is that QF is showing as version 4.3.1 (same as before), whereas you claim this should be 4.17.5. But I installed using the normal mechanism via the TB Tools menu invoking TB Add-ons Manager..

The only other add-ons I left extant were Mail Redirect and Lightning, which I assume ought not to have interfered in any way.

So having gone though this exercise, I am back to square one, with exactly the same fault as before; in fact, making absolutely no difference to versions or anything.

**Comment by Dimathman** [comment]
Addendum to the above: I can't tell what version of FiltaQuilla I have installed, as per your comment, because I can't access the preferences to tell - catch 22. But I do have some version installed and working because I can see some default additions in the Filters menus.

**Comment by RealRaven2000** [comment]
> Puzzling though, is that QF is showing as version 4.3.1 (same as before), whereas you claim this should be 4.17.5. But I installed using the normal mechanism via the TB Tools menu invoking TB Add-ons Manager..

ok, that's my own fault. that's actually the current version of QuickFolders (QF) - I abbreviate quickFilters as qF and 4.3.1 is the correct version. What version of Thunderbird are you running now? Which OS?

Check, if you get any error messages when you expand tools > Add-on options sub menu for the first time.

If you go to Add-ons Manager and click on the FiltaQuilla entry you should see a detail page with version number:

![image](https://user-images.githubusercontent.com/6314674/72159200-c1189e00-33b3-11ea-8f98-49182c1ace67.png)

**Comment by Dimathman** [comment]
The version showing for qF (QuickFilters) is 4.3.1. I am not really interested in QuickFolders.
The version of FiltaQuilla that I installed, and is apparently running OK with default preferences, is 2.0. But I can only deduce this from the source download page - obviously not from preferences.
The version of TB is 68.3.1 (64 bit). It only auto-updated to this 2 days ago.
My Mac OS is 10.10.5 (Yosemite). Your add-ons claim to be OK from 10.9 upwards.
The Tools/Preferences menu is showing nothing at all - not even the greyed out "No options available" message. I might try installing some other add-on that has options and see if I am offered them via this menu.

**Comment by RealRaven2000** [comment]
> The version of FiltaQuilla that I installed, and is apparently running OK with default preferences, is 2.0. But I can only deduce this from the source download page - obviously not from preferences.

the download page? what about Add-on Manager / Extensions and then click on the (installed) FiltaQuilla extension (if it is installed its not the download page, but still within Add-ons Manager (no separate tab), looking like this:

![image](https://user-images.githubusercontent.com/6314674/72168675-7f90ee80-33c5-11ea-8518-487b96f7d0d0.png)

is that the page you mean?

**Comment by Dimathman** [comment]
Yes, sorry, I meant via the "add-ons manager" page, both the page it was installed from, and the Extensions page, exactly as you cite above. It is showing Version 2.0, last updated 10 January 2020.

**Comment by RealRaven2000** [comment]
ok, that means it (correct version) is definitely installed and not disabled, onwards and upwards! Can you try and create an empty test profile to see if it isn't your production profile or some weird setting / other Add-ons / plugins interfering. Also, to rule out locale problems - what language is your Thunderbird in?

**Comment by Dimathman** [comment]
OK, but this is more than a trivial exercise, particularly when I need to take care not to disrupt my normal business too much. I may be a day or so responding. My Thunderbird in normal English.

**Comment by trwww** [comment]
> Can you try and create an empty test profile to see if it isn't your production profile or some weird setting / other Add-ons / plugins interfering.

Not gonna hurt for someone else to try, but note that I already did this in https://github.com/RealRaven2000/FiltaQuilla/issues/37#issuecomment-566782626

**Comment by Dimathman** [comment]
Now 7 months on, and have tried everything. New versions of TB since then. FiltaQuilla is still not working. No preferences showing. All I really want is regular expressions in the filter fields, and although these appear by default, and appear to work, many copies of these FQ fields appear in the menu after some time. Shame because FQ was a great add-on in the distant past, but not any more ....
